### DISCUSSION TITLE
Shorter magic links

### DISCUSSION THREAD

**Discussion by wild-lotus** [pbteja1998/remix-auth-email-link#33] [Ideas]
Long links look scary, so I think we could make the generated magic link as short as possible, to make it as less intimidating as possible. Magic link is the entry point for an app, so UX is very important!

Make make a shorter link, we need a shorter payload. Some ideas for optimizations:
- The email address is included twice, top level and inside `form`.
- Ability to manipulate if/how the `form` is included in the payload. I might just want to get something to the server, but not need it in the magic link payload.
- Make keys shorter. Specially `validateSessionMagicLink` is quite long!

What do you think?

**Comment by pbteja1998** [comment]
Good ideas. Please feel free to open a PR with these changes. Thanks.

**Reply by wild-lotus** [reply]
I created https://github.com/pbteja1998/remix-auth-email-link/pull/34, I am leaving `form` manipulation for a different PR.

**Reply by pbteja1998** [reply]
Merged the PR. Thank you! You can find it in `v2.0.0`

**Comment by wild-lotus** [comment]
btw, I think this also affects the spam score, the shorter the link the better!
